### PR TITLE
Spruce up PIC theory section

### PR DESCRIPTION
**Pull request by eebasso**
This PR fixes the citation, figure, and equation references. It also ensures the usual alignment for displayed math equations. Other editorial changes are included.

### CODE REVIEW

File: Docs/source/theory/picsar_theory.rst [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
 
-In the *electromagnetic particle-in-cell method* (Birdsall and Langdon 1991),
+In the *electromagnetic particle-in-cell method* :cite:p:`pt-Birdsalllangdon`,
```
Comment (ax3l) [inline Docs/source/theory/picsar_theory.rst]: We could also cite Hockney-Eastwood here, both are the two standard books.

File: Docs/source/theory/picsar_theory.rst [hunk 2]
```diff
@@ -1,36 +1,34 @@
-.. raw:: latex
```
Comment (ax3l) [inline Docs/source/theory/picsar_theory.rst:1]: Can you rename this file to `pic.rst`?
The picsar name is historic and not used anymore for the core routines.

Comment (eebasso) [reply Docs/source/theory/picsar_theory.rst:1]: Sure

File: Docs/source/theory/picsar_theory.rst (hunk 1 above)
Comment (eebasso) [reply Docs/source/theory/picsar_theory.rst]: Okay

**Review by ax3l — APPROVED** [review]

**Pull request merged**